react-file-uploads/server/index.js

```diff
@@ -0,0 +1,65 @@
+require('dotenv/config');
+const pg = require('pg');
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+const uploadsMiddleware = require('./uploads-middleware');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/uploads', uploadsMiddleware, (req, res, next) => {
+  const { caption } = req.body;
+  if (!caption) {
+    throw new ClientError(400, 'caption is a required field');
+  }
+  // v--- my code ---v
+  //* - create a url for the image by combining '/images' with req.file.filename
+  const url = '/images/' + req.file.filename;
+  //* - insert the "caption" and "url" into the "images" table
+  const obj = {
+    caption,
+    url
+  };
+  const sql = `
+    insert into "images"("caption", "url")
+    values($1, $2)
+    returning *
+  `;
+  // * - respond with the inserted row data
+  db.query(sql, obj)
+    .then(result => {
+      res.json(result.rows);
```

Comment:
You want to send back the single object, not the object inside an array so it would be better to do:
```JAVASCRIPT
res.json(result.rows[0]);
```

Reply:
noted, thank you!